[android-auto] Fix turn directions

AndrewShkrob opened the pull request:
Current street replaced with next street
Next street replaced with next next street

vng commented on routing/following_info.hpp:
```diff
@@ -73,6 +73,8 @@ class FollowingInfo
   std::string m_targetName;
   // Street name to display. May be empty.
   std::string m_displayedStreetName;
```
I see one strange thing here (except the vars' names - this is a separate issue).

AFAI understand, m_displayedStreetName == m_targetName with some processing?
We use m_displayedStreetName (not m_targetName) constructing android.RoutingInfo
Should we do the same for m_nextStreetName?

vng commented on routing/routing_session.cpp:
```diff
@@ -417,11 +417,13 @@ void RoutingSession::GetRouteFollowingInfo(FollowingInfo & info) const
   m_route->GetNextTurnStreetName(targetRoadNameInfo);
   GetFullRoadName(targetRoadNameInfo, info.m_targetName);
+  m_route->GetNextNextTurnStreetName(nextRoadNameInfo);
+  GetFullRoadName(nextRoadNameInfo, info.m_nextStreetName);
```
As I mentioned above, here we assign m_nextStreetName the same way as m_targetName.

But in Android code we use m_displayedStreetName.

vng commented on android/app/src/main/cpp/app/organicmaps/Framework.cpp:
```diff
@@ -1252,7 +1252,7 @@ Java_app_organicmaps_Framework_nativeGetRouteFollowingInfo(JNIEnv * env, jclass)
   static jmethodID const ctorRouteInfoID =
```
Update the ctor comment above?

vng commented:
Is it some kind of the AA standard to display next next street or our invention? :)

biodranik commented:
@vng I understanding it like this (@AndrewShkrob please correct if I'm wrong):

On AA there is a nav template with two named fields that can be filled by apps:
1. Current street
2. Next street

In AA terms, the current street is not the street where a user is driving now, but the currently visible direction (e.g. the upcoming turn, the _next_ street after the one where a car is moving now).

The next street in AA terms is the secondary direction, the second turn after the first turn (e.g. "turn left in 1km" is the "current" in AA terms, and "after that, in 100m keep left" is the "next" in AA terms).

biodranik commented on android/app/src/main/java/app/organicmaps/routing/RoutingInfo.java:
```diff
@@ -24,6 +24,8 @@ public class RoutingInfo
   public final String currentStreet;
```
Does currentStreet contain the road where the car is moving now?

AndrewShkrob replied:
Yes

AndrewShkrob commented:
I have updated the code and variable names to make them more obvious.

Previously we had `info.m_displayedStreetName = info.m_targetName;`

I removed `m_displayedStreetName`
Now we have
```cpp
  // Current street name.
  std::string m_currentStreetName;
  // The next street name. May be empty.
  std::string m_nextStreetName;
  // The next next street name. May be empty.
  std::string m_nextNextStreetName;
```

AndrewShkrob commented:
> @vng I understanding it like this (@AndrewShkrob please correct if I'm wrong):
> 
> On AA there is a nav template with two named fields that can be filled by apps:
> 
> 1. Current street
> 2. Next street
> 
> In AA terms, the current street is not the street where a user is driving now, but the currently visible direction (e.g. the upcoming turn, the _next_ street after the one where a car is moving now).
> 
> The next street in AA terms is the secondary direction, the second turn after the first turn (e.g. "turn left in 1km" is the "current" in AA terms, and "after that, in 100m keep left" is the "next" in AA terms).

https://developers.google.com/cars/design/create-apps/apps-for-drivers/templates/navigation-template

<table>
  <tbody>
    <tr>
      <td markdown>
<img src="https://github.com/user-attachments/assets/afbcfc23-a387-438b-8a65-6eea35ace79c"/>
      </td>
      <td markdown>

When the routing card is in routing state (as opposed to [message state](https://developers.google.com/cars/design/create-apps/apps-for-drivers/templates/navigation-template?hl=en#message-state)), it shows the following information:

1. Current step: includes icon (typically a direction arrow), distance, and cue text (which may include image spans such as route markers)
2. Lane guidance (optional): shown either as simple lane-assist images or as a larger junction image (flexible size with maximum height of 200dp)
3. Next step (optional): includes icon and cue, and can appear only at the bottom of a routing card that doesn’t include a junction image
      </td>
    </tr>
  </tbody>
</table>

So, it is called `current step` but it actually displays next street name.
To be more precise, it is intended to display not just the street name but a cue, like "In 500 meters turn left to the `street name`"

biodranik commented on routing/following_info.hpp:
```diff
@@ -68,11 +68,11 @@ class FollowingInfo
   // depends on the number of notifications to prononce.
   std::vector<std::string> m_turnNotifications;
   // Current street name.
```
```suggestion
  // Current street name. May be empty.
```

Right?

biodranik reviewed (approved):
Thanks! A minor question
